src/test/java/org/mifos/integrationtest/cucumber/stepdef/VoucherManagementStepDef.java

```diff
@@ -446,21 +452,25 @@ public void iCanAssertThatRedemptionWasUnsuccessfulByAssertingTheStatusInRespons
 
     @Then("I should be able to assert response body from callback on {string}")
     public void iShouldBeAbleToAssertResponseBodyFromCallback(String endpoint) {
-        verify(putRequestedFor(urlEqualTo(endpoint)).withRequestBody(matchingJsonPath("$.registerRequestId", equalTo(requestId))));
-        verify(putRequestedFor(urlEqualTo(endpoint)).withRequestBody(matchingJsonPath("$.numberFailedCases", equalTo("0"))));
+        await().atMost(awaitMost, SECONDS).untilAsserted(() -> {
+            verify(putRequestedFor(urlEqualTo(endpoint)).withRequestBody(matchingJsonPath("$.registerRequestId", equalTo(requestId))));
+            verify(putRequestedFor(urlEqualTo(endpoint)).withRequestBody(matchingJsonPath("$.numberFailedCases", equalTo("0"))));
+        });
     }
 
     @Then("I will call the fetch voucher API with expected status of {int}")
     public void iWillCallTheFetchVoucherAPIWithExpectedStatusOf(int responseCode) {
-        RequestSpecification requestSpec = Utils.getDefaultSpec();
-        scenarioScopeState.response = RestAssured.given(requestSpec).header("Content-Type", "application/json")
-                .header("X-Registering-Institution-ID", registeringInstitutionId)
-                .baseUri(voucherManagementConfig.voucherManagementContactPoint).expect()
-                .spec(new ResponseSpecBuilder().expectStatusCode(responseCode).build()).when()
-                .get(voucherManagementConfig.fetchVoucherEndpoint + "/" + serialNumber).andReturn().asString();
-
-        fetchVoucherResponseBody = scenarioScopeState.response;
-        logger.info("Voucher Response: {}", scenarioScopeState.response);
+        await().atMost(awaitMost, SECONDS).untilAsserted(() -> {
```

Comment:
here await may not be needed. Please check

Reply:
Fetch voucher API test requires 3 second of wait.